lib/ExpensiMark.js

```diff
@@ -346,6 +348,40 @@ export default class ExpensiMark {
         return replacedText;
     }
 
+    /**
+     * Map the block to newline
+     *
+     * @param {String} htmlString
+     *
+     * @returns {String}
+     */
+    mapBlock(htmlString) {
+        const splitText = htmlString.split('[block]');
+        let joinedText = '';
+
+        // delete empty spaces at the end
+        while (splitText.length) {
+            if (splitText[splitText.length - 1].trim().length === 0 && !splitText[splitText.length - 1].match(/\n/)) {
+                splitText.pop();
+            } else {
+                break;
+            }
+        }
+        for (let index = 0; index < splitText.length - 1; index++) {
```

Comment:
I don't think you're using the index. The same could be achieved with a `forEach` loop.
What do you think? Won't it make it a bit easier to read?

Reply:
i use it to avoid adding line on last index but that's ok we can do it with forEach also